## Actividad 1

### Objetivo de la actividad:
El laboratorio de geoespacial del CIC busca una nueva forma de dar a conocer los resultados sobre los datos de Covid-19 en el terrotorio de la CDMX. Por ello, se haran intentos de eleborar mapas iteractivos usadon Shiny, una herramienta que permite generar dashboards en R.


Se pretende implimentar primero un mapa interactivo basandose en los datos provenientes de la Secretaria de SALUD, para acceder a tales registros ingresar al siguiente [enlace](https://www.gob.mx/salud/documentos/datos-abiertos-152127). Además se basara parcialmente en el siguiente repositorio, [enlace](https://github.com/medinaalonso/Mapas-COVID-19-M-xico). Tal repositorio fue elaborado por un colaborador del laboratorio del CIC.

### Elaboración de la actividad:

Como inicio de la actividad. Se implementara parcialmente el codigo del rpositorio presentado anteriormente. Se presentaran los pasos en caso de que se deseé implementar:

1. Descargar el documento del enlace presentado anteriormente. Y ubicarlo en la carpeta donde esta este Jupyter Notebook. En este ejemplo se tomara como ejemplo los datos historicos de la fecha más reciente, que es la del 31 de Enero del 2021, presentes en el siguiente [enlace](https://www.gob.mx/cms/uploads/attachment/file/613630/Datos_abiertos_hist_ricos_2021.pdf) . En este caso se coloco en una carpeta nombrada __docs__ y se recombro el archivo por __COVID19MEXICO.csv__ Resultado de la actividad uno:

![Ubicación del archivo a trabajar](./img/img1.PNG)


En caso de replicar este ejercicio favor de dirigirse al enlace anterior, descargar el docuemnto, colocarlo en la carpeta __docs__ , descomprimirlo y cambiarle el nombre o modificar esa parte del codigo. _Este archivo no esta presente en el repositorio debido a su peso_.


2. Cargamos el archivo __COVID19MEXICO.csv__ y se ejecuta la primera celda. Esta celda brinda las coordenadas de los distintos Estados de la Republica Mexicana en un nuevo documenteo .csv. Se baso parcialmente el repositorio presentado anteriormente.


In [2]:
# Importanciones necesarias
import pandas as pd #Para manipulación de pandas

import numpy as np #Para manipulación de numpy

In [7]:
# Lectura de .csv
df = pd.read_csv("docs/COVID19MEXICO.csv")

In [8]:
# Revision de .csv
df.head()

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
0          2021-01-31      z482b8       1      12           9     2   
1          2021-01-31      z49a69       1      12          23     1   
2          2021-01-31      z23d9d       1      12          22     2   
3          2021-01-31      z24953       1      12           9     1   
4          2021-01-31      zz8e77       1      12           9     2   

   ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ... OTRO_CASO  \
0            9            9             12              1  ...         2   
1           23           23              4              2  ...         1   
2           24           22              9              1  ...         2   
3            9            9             10              1  ...         1   
4            9            9              2              1  ...         2   

  TOMA_MUESTRA_LAB RESULTADO_LAB  TOMA_MUESTRA_ANTIGENO  RESULTADO_ANTIGENO  \
0                2            97                      2                  97   
1                2            97                      2                  97   
2                2            97                      2                  97   
3                1             2                      2                  97   
4                2            97                      2                  97   

   CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
0                    1        99             México           97   97  
1                    2        99             México           97    1  
2                    6        99             México           97   97  
3                    7        99             México           97   97  
4                    6        99             México           97   97  

[5 rows x 40 columns]

In [9]:
# Revisamos el tamaño de. csv
df.shape

(4700463, 40)

Se observa que el documento .csv tiene 4,700,463 millones de registros y 40 campos.

In [10]:
# Observamos la informacion del DataFrame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4700463 entries, 0 to 4700462
Data columns (total 40 columns):
 #   Column                 Dtype 
---  ------                 ----- 
 0   FECHA_ACTUALIZACION    object
 1   ID_REGISTRO            object
 2   ORIGEN                 int64 
 3   SECTOR                 int64 
 4   ENTIDAD_UM             int64 
 5   SEXO                   int64 
 6   ENTIDAD_NAC            int64 
 7   ENTIDAD_RES            int64 
 8   MUNICIPIO_RES          int64 
 9   TIPO_PACIENTE          int64 
 10  FECHA_INGRESO          object
 11  FECHA_SINTOMAS         object
 12  FECHA_DEF              object
 13  INTUBADO               int64 
 14  NEUMONIA               int64 
 15  EDAD                   int64 
 16  NACIONALIDAD           int64 
 17  EMBARAZO               int64 
 18  HABLA_LENGUA_INDIG     int64 
 19  INDIGENA               int64 
 20  DIABETES               int64 
 21  EPOC                   int64 
 22  ASMA                   int64 
 23  INMUSUP

### Agregar Latitud y Longitud.

Para agregar la `Longitud` y `Latitud` en el DataFrame, es necesario de usar la columna de _` ENTIDAD_RES `_. 

In [11]:
# Se copia la columna "ENTIDAD_RES" para generar las columnas "LONGITUD" y "LATITUD".
df['LONGITUD'] = df['ENTIDAD_RES']
df['LATITUD'] = df['ENTIDAD_RES']

# Diccionario para sustituir en la columna "ENTIDAD_RES"
dic_remplazar_nom = {
    1:'AGUASCALIENTES',
    2:'BAJA_CALIFORNIA',
    3:'BAJA_CALIFORNIA_SUR',
    4:'CAMPECHE',
    5:'COAHUILA_DE_ZARAGOZA',
    6:'COLIMA',
    7:'CHIAPAS',
    8:'CHIHUAHUA',
    9:'CIUDAD_DE_MEXICO',
    10:'DURANGO',
    11:'GUANAJUATO',
    12:'GUERRERO',
    13:'HIDALGO',
    14:'JALISCO',
    15:'MEXICO',
    16:'MICHOACAN_DE_OCAMPO',
    17:'MORELOS',
    18:'NAYARIT',
    19:'NUEVO_LEON',
    20:'OAXACA',
    21:'PUEBLA',
    22:'QUERETARO',
    23:'QUINTANA_ROO',
    24:'SAN_LUIS_POTOSI',
    25:'SINALOA',
    26:'SONORA',
    27:'TABASCO',
    28:'TAMAULIPAS',
    29:'TLAXCALA',
    30:'VERACRUZ_DE_IGNACIO_DE_LA_LLAVE',
    31:'YUCATAN',
    32:'ZACATECAS'
}

# Diccionario para sustituir en la columna "LONGITUD"
dic_remplazar_lon = {
    1:-102.35861,
    2:-115.99649,
    3:-111.82525,
    4:-90.37322,
    5:-103.18592,
    6:-103.72244,
    7:-93.11121,
    8:-106.21278,
    9:-99.1276627,
    10:-104.53320,
    11:-101.25289,
    12:-99.46573,
    13:-103.47140,
    14:-103.34616,
    15:-99.70051,
    16:-101.26428,
    17:-99.30873,
    18:-105.08841,
    19:-100.31370,
    20:-96.76070,
    21:-98.18664,
    22:-100.45296,
    23:-88.3038100,
    24:-101.23564,
    25:-107.87211,
    26:-110.96133,
    27:-92.9302826,
    28:-99.14599,
    29:-98.23010,
    30:-96.83564,
    31:-89.56206,
    32:-102.58325
}

# Diccionario para sustituir en la columna "LATITUD"
dic_remplazar_lat = {
    1:21.77111,
    2:30.73563,
    3:25.11662,
    4:19.48657,
    5:25.23892,
    6:19.14736,
    7:16.71165,
    8:28.15096,
    9:19.4284706,
    10:24.05718,
    11:21.07784,
    12:17.36164,
    13:20.79916,
    14:20.67663,
    15:19.35601,
    16:19.69950,
    17:18.94315,
    18:21.69107,
    19:25.67693,
    20:17.11867,
    21:18.95652,
    22:20.72682,
    23:18.51413,
    24:22.24988,
    25:25.69983,
    26:29.08919,
    27:17.9868908,
    28:23.74174,
    29:19.32226,
    30:19.57105,
    31:21.04023,
    32:22.77093
}

# Usar los diccionarios para transformar los diccionarios
df1 = df.replace({'ENTIDAD_RES': dic_remplazar_nom,'LONGITUD': dic_remplazar_lon,'LATITUD': dic_remplazar_lat})

# Guardar datos en la carpeta "docs"
df1.to_csv("docs/COVID19MEXICO_datos_modificados.csv", index=False)

# Observamos el resultado
df1.head()

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
0          2021-01-31      z482b8       1      12           9     2   
1          2021-01-31      z49a69       1      12          23     1   
2          2021-01-31      z23d9d       1      12          22     2   
3          2021-01-31      z24953       1      12           9     1   
4          2021-01-31      zz8e77       1      12           9     2   

   ENTIDAD_NAC       ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
0            9  CIUDAD_DE_MEXICO             12              1  ...   
1           23      QUINTANA_ROO              4              2  ...   
2           24         QUERETARO              9              1  ...   
3            9  CIUDAD_DE_MEXICO             10              1  ...   
4            9  CIUDAD_DE_MEXICO              2              1  ...   

  RESULTADO_LAB TOMA_MUESTRA_ANTIGENO RESULTADO_ANTIGENO  CLASIFICACION_FINAL  \
0            97                     2                 97                    1   
1            97                     2                 97                    2   
2            97                     2                 97                    6   
3             2                     2                 97                    7   
4            97                     2                 97                    6   

   MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI    LONGITUD    LATITUD  
0        99             México           97   97  -99.127663  19.428471  
1        99             México           97    1  -88.303810  18.514130  
2        99             México           97   97 -100.452960  20.726820  
3        99             México           97   97  -99.127663  19.428471  
4        99             México           97   97  -99.127663  19.428471  

[5 rows x 42 columns]

### Generación de dataset para R-Studio

Para generar el dataset que se usara para generar el mapa en el Shiny dashboard, es necesario hacer una agrupacion de datos y una modificacion para hacerlo y observar el numero total de casos en la __Republica Mexica hasta el 31 de Enero del 2021__. 

In [12]:
# Se extrae los datos del total de casos por estado, contando los datos de latitud y longitud.
serie_entidad = df1['ENTIDAD_RES'].value_counts()
serie_latitud = df1['LATITUD'].value_counts()
serie_longitud = df1['LONGITUD'].value_counts()

# Se guardan los datos extraidos en un array
estado = serie_entidad.index.tolist()
latitud = serie_latitud.index.tolist()
longitud = serie_longitud.index.tolist()
casos = serie_entidad.tolist()

# Se guardan los datos extraidos en un diccionario
dic_mapa_1 = {
    'estado':estado,
    'latitude':latitud,
    'longitude':longitud,
    'cases':casos
}

# Se genera un DataFrame con el diccionario
df_mapa_1 = pd.DataFrame(dic_mapa_1)

# Se almacena el documento el la carpeta "map_docs"
df_mapa_1.to_csv("map_docs/mapa_Mexico_total_de_casos.csv",index=False)

# Se observa el DataFrame resultante
df_mapa_1.head()

estado   latitude   longitude    cases
0  CIUDAD_DE_MEXICO  19.428471  -99.127663  1461114
1            MEXICO  19.356010  -99.700510   520107
2        GUANAJUATO  21.077840 -101.252890   231875
3        NUEVO_LEON  25.676930 -100.313700   228617
4           JALISCO  20.676630 -103.346160   160632

### Modificación del DataSet para ver si murio o no murio la persona

Con la columna `FECHA_DEF` nos damos idea si la persona fallecio o no. 


Si la fecha es __9999-99-99__ significa que fallecio y por lo tanto se sustituira con un __1__, que significa que _ Murio_. En otro caso se sustituira con un __2__.

In [13]:
#Se hace copia del dataset
df2 = df1.copy()

In [14]:
# Exploracion de los datos de "FECHA_DEF"
df2.iloc[:,12].value_counts()

9999-99-99    4485163
2021-01-19       1323
2021-01-20       1313
2021-01-17       1288
2021-01-18       1287
               ...   
2020-01-06          3
2021-01-31          3
2020-01-05          2
2020-01-02          1
2020-01-04          1
Name: FECHA_DEF, Length: 397, dtype: int64

In [15]:
# Se sustituyen los valores que no son 9999-99-99 por 1
df2['FECHA_DEF'] = df2['FECHA_DEF'].map(lambda x : x if x == '9999-99-99' else 1).copy()

# Se sustituyen los valores que son 9999-99-99 por 2
df2['FECHA_DEF'] = df2['FECHA_DEF'].replace({'9999-99-99':2})

# Se renombra la columna de fecha de desenso por si esta vivo o no 
df2.rename(columns={'FECHA_DEF':'ESTADO_PACIENTE'})

# Observacion de los cambios
df2['FECHA_DEF'].value_counts()

2    4485163
1     215300
Name: FECHA_DEF, dtype: int64

### Coordenadas de las alcaldias de la CDMX

Como ultimo paso en la generación de un DataSet con los datos de la CDMX y sus alcaldias. Se sustitulleron los numeros de las alcaldias por su respectivo nombre

In [16]:
# Se copio el DataSet a trabajar
df3 = df2.copy()

In [17]:
# Se observa sus primeros registros
df3.head()

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
0          2021-01-31      z482b8       1      12           9     2   
1          2021-01-31      z49a69       1      12          23     1   
2          2021-01-31      z23d9d       1      12          22     2   
3          2021-01-31      z24953       1      12           9     1   
4          2021-01-31      zz8e77       1      12           9     2   

   ENTIDAD_NAC       ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
0            9  CIUDAD_DE_MEXICO             12              1  ...   
1           23      QUINTANA_ROO              4              2  ...   
2           24         QUERETARO              9              1  ...   
3            9  CIUDAD_DE_MEXICO             10              1  ...   
4            9  CIUDAD_DE_MEXICO              2              1  ...   

  RESULTADO_LAB TOMA_MUESTRA_ANTIGENO  RESULTADO_ANTIGENO  \
0            97                     2                  97   
1            97                     2                  97   
2            97                     2                  97   
3             2                     2                  97   
4            97                     2                  97   

   CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  \
0                    1        99             México           97   97   
1                    2        99             México           97    1   
2                    6        99             México           97   97   
3                    7        99             México           97   97   
4                    6        99             México           97   97   

     LONGITUD    LATITUD  
0  -99.127663  19.428471  
1  -88.303810  18.514130  
2 -100.452960  20.726820  
3  -99.127663  19.428471  
4  -99.127663  19.428471  

[5 rows x 42 columns]

In [18]:
# Se genera un filtro con los datos donde su ubicacion sea la CDMX
fil_1 = df3['ENTIDAD_RES']=='CIUDAD_DE_MEXICO'

# Se aplica el filtro y se guarda
df4 = df3[fil_1].copy()

In [19]:
# Se observan los valores de las alcaldias
df4['MUNICIPIO_RES'].value_counts()

7      242039
5      165098
10     143402
12     125097
13      87380
3       86750
15      85874
11      82813
17      76306
8       66258
2       63573
14      59205
6       58961
16      52631
9       34900
4       30809
999        12
106         6
Name: MUNICIPIO_RES, dtype: int64

Como se puede observar los siguientes registros no pertenecen a ninguna Alcaldia registrada en el Diccionario del documento .csv. 


Esos datos son: __999__ y __106__, por ello se optaron por eliminar.

In [20]:
# Se transformaron los valores 999 y 106 por NULL
df4['MUNICIPIO_RES'] = df4['MUNICIPIO_RES'].map(lambda x : x if (x != 999) and (x != 106) else None).copy()

# Se eliminaron los nulos
df4 = df4.dropna()

# Se comprueba la operacion
df4.isnull().sum()

FECHA_ACTUALIZACION      0
ID_REGISTRO              0
ORIGEN                   0
SECTOR                   0
ENTIDAD_UM               0
SEXO                     0
ENTIDAD_NAC              0
ENTIDAD_RES              0
MUNICIPIO_RES            0
TIPO_PACIENTE            0
FECHA_INGRESO            0
FECHA_SINTOMAS           0
FECHA_DEF                0
INTUBADO                 0
NEUMONIA                 0
EDAD                     0
NACIONALIDAD             0
EMBARAZO                 0
HABLA_LENGUA_INDIG       0
INDIGENA                 0
DIABETES                 0
EPOC                     0
ASMA                     0
INMUSUPR                 0
HIPERTENSION             0
OTRA_COM                 0
CARDIOVASCULAR           0
OBESIDAD                 0
RENAL_CRONICA            0
TABAQUISMO               0
OTRO_CASO                0
TOMA_MUESTRA_LAB         0
RESULTADO_LAB            0
TOMA_MUESTRA_ANTIGENO    0
RESULTADO_ANTIGENO       0
CLASIFICACION_FINAL      0
MIGRANTE                 0
P

In [21]:
# Se copia la columna "MUNICIPIO_RES" para generar las columnas "LONGITUD_CDMX" y "LATITUD_CDMX".
df4['LONGITUD_CDMX'] = df4['MUNICIPIO_RES'].astype(int)
df4['LATITUD_CDMX'] = df4['MUNICIPIO_RES'].astype(int)
df4['MUNICIPIO_RES'] = df4['MUNICIPIO_RES'].astype(int)

# Diccionario con nombre de alcaldias CDMX
diccionario_CDMX_Alcadias_nombres={
    2:'AZCAPOTZALCO',
    3:'COYOACAN',
    4:'CUAJIMALPA_DE_MORELOS',
    5:'GUSTAVO_A._MADERO',
    6:'IZTACALCO',
    7:'IZTAPALAPA',
    8:'LA_MAGDALENA_CONTRERAS',
    9:'MILPA_ALTA',
    10:'ALVARO_OBREGON',
    11:'TLAHUAC',
    12:'TLALPAN',
    13:'XOCHIMILCO',
    14:'BENITO_JUAREZ',
    15:'CUAUHTEMOC',
    16:'MIGUEL HIDALGO',
    17:'VENUSTIANO CARRANZA'
}

# Diccionario con longitud de las alcaldias CDMX
diccionario_CDMX_Alcadias_longitud={
    2:-99.1859436,
    3:-99.1617432,
    4:-99.2908936,
    5:-99.1107483,
    6:-99.0977783,
    7:-99.0622406,
    8:-99.2111816,
    9:-99.023056,
    10:-99.2032928,
    11:-99.0050735,
    12:-99.1620636,
    13:-99.1035614,
    14:-99.1576614,
    15:-99.1461182,
    16:-99.2002411,
    17:-99.1049881
}

# Diccionario con latitud de las alcaldias CDMX
diccionario_CDMX_Alcadias_latitud={
    2:19.4869804,
    3:19.3467007,
    4:19.3691998,
    5:19.4939194,
    6:19.3952808,
    7:19.3552895,
    8:19.3321209,
    9:19.192222,
    10:19.3586693,
    11:19.28689,
    12:19.2951298,
    13:19.2546501,
    14:99.1576614,
    15:19.3983994,
    16:19.4341106,
    17:19.4436092
}

# Aplicar transformacion de los diccionarios
df5 = df4.replace({'MUNICIPIO_RES': diccionario_CDMX_Alcadias_nombres,'LATITUD_CDMX':diccionario_CDMX_Alcadias_latitud,'LONGITUD_CDMX':diccionario_CDMX_Alcadias_longitud}).copy()

# Guardar datos en la carpeta "docs"
df5.to_csv("docs/COVID19MEXICO_datos_CDMX.csv", index=False)

# Comprobar operaciones
df5.head()

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
0          2021-01-31      z482b8       1      12           9     2   
3          2021-01-31      z24953       1      12           9     1   
4          2021-01-31      zz8e77       1      12           9     2   
6          2021-01-31      z2d0c4       1      12           9     1   
7          2021-01-31      z26b82       1      12           9     1   

   ENTIDAD_NAC       ENTIDAD_RES   MUNICIPIO_RES  TIPO_PACIENTE  ...  \
0            9  CIUDAD_DE_MEXICO         TLALPAN              1  ...   
3            9  CIUDAD_DE_MEXICO  ALVARO_OBREGON              1  ...   
4            9  CIUDAD_DE_MEXICO    AZCAPOTZALCO              1  ...   
6            9  CIUDAD_DE_MEXICO       IZTACALCO              1  ...   
7            9  CIUDAD_DE_MEXICO      IZTAPALAPA              1  ...   

  RESULTADO_ANTIGENO CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  \
0                 97                   1        99             México   
3                 97                   7        99             México   
4                 97                   6        99             México   
6                 97                   7        99             México   
7                  2                   7        99             México   

   PAIS_ORIGEN  UCI   LONGITUD    LATITUD  LONGITUD_CDMX  LATITUD_CDMX  
0           97   97 -99.127663  19.428471     -99.162064     19.295130  
3           97   97 -99.127663  19.428471     -99.203293     19.358669  
4           97   97 -99.127663  19.428471     -99.185944     19.486980  
6           97   97 -99.127663  19.428471     -99.097778     19.395281  
7           97   97 -99.127663  19.428471     -99.062241     19.355290  

[5 rows x 44 columns]

In [23]:
df6 = df5.copy()

In [26]:
df6.head()

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
0          2021-01-31      z482b8       1      12           9     2   
3          2021-01-31      z24953       1      12           9     1   
4          2021-01-31      zz8e77       1      12           9     2   
6          2021-01-31      z2d0c4       1      12           9     1   
7          2021-01-31      z26b82       1      12           9     1   

   ENTIDAD_NAC       ENTIDAD_RES   MUNICIPIO_RES  TIPO_PACIENTE  ...  \
0            9  CIUDAD_DE_MEXICO         TLALPAN              1  ...   
3            9  CIUDAD_DE_MEXICO  ALVARO_OBREGON              1  ...   
4            9  CIUDAD_DE_MEXICO    AZCAPOTZALCO              1  ...   
6            9  CIUDAD_DE_MEXICO       IZTACALCO              1  ...   
7            9  CIUDAD_DE_MEXICO      IZTAPALAPA              1  ...   

  RESULTADO_ANTIGENO CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  \
0                 97                   1        99             México   
3                 97                   7        99             México   
4                 97                   6        99             México   
6                 97                   7        99             México   
7                  2                   7        99             México   

   PAIS_ORIGEN  UCI   LONGITUD    LATITUD  LONGITUD_CDMX  LATITUD_CDMX  
0           97   97 -99.127663  19.428471     -99.162064     19.295130  
3           97   97 -99.127663  19.428471     -99.203293     19.358669  
4           97   97 -99.127663  19.428471     -99.185944     19.486980  
6           97   97 -99.127663  19.428471     -99.097778     19.395281  
7           97   97 -99.127663  19.428471     -99.062241     19.355290  

[5 rows x 44 columns]

### Generación de dataset para R-Studio

Para generar el dataset que se usara para generar el mapa en el Shiny dashboard, es necesario hacer una agrupacion de datos y una modificacion para hacerlo y observar el numero total de casos en la __Ciudad de Mexico hasta el 31 de Enero del 2021__. 

In [28]:
# Se extrae los datos del total de casos por estado, contando los datos de latitud y longitud.
serie_entidad_cdmx = df6['MUNICIPIO_RES'].value_counts()
serie_latitud_cdmx = df6['LATITUD_CDMX'].value_counts()
serie_longitud_cdmx = df6['LONGITUD_CDMX'].value_counts()

# Se guardan los datos extraidos en un array
alcaldias_cdmx = serie_entidad_cdmx.index.tolist()
latitud_cdmx = serie_latitud_cdmx.index.tolist()
longitud_cdmx = serie_longitud_cdmx.index.tolist()
casos_cdmx = serie_entidad_cdmx.tolist()

# Se guardan los datos extraidos en un diccionario
dic_mapa_2 = {
    'alcaldias':alcaldias_cdmx,
    'latitude':latitud_cdmx,
    'longitude':longitud_cdmx,
    'cases':casos_cdmx
}

# Se genera un DataFrame con el diccionario
df_mapa_2 = pd.DataFrame(dic_mapa_2)

# Se almacena el documento el la carpeta "map_docs"
df_mapa_2.to_csv("map_docs/mapa_CDMX_total_de_casos.csv",index=False)

# Se observa el DataFrame resultante
df_mapa_2.head()

alcaldias   latitude  longitude   cases
0         IZTAPALAPA  19.355290 -99.062241  242039
1  GUSTAVO_A._MADERO  19.493919 -99.110748  165098
2     ALVARO_OBREGON  19.358669 -99.203293  143402
3            TLALPAN  19.295130 -99.162064  125097
4         XOCHIMILCO  19.254650 -99.103561   87380